### PR TITLE
Add spec completions

### PR DESCRIPTION
**Pull request by arichiardi**
Here it is, spec completions are working fine.

I need some more time if we want to setup a couple of tests, but it looks like it is already working fine. One thing to think about is the position of the completion candidates. I put the before everything at the moment.

Any comment/feedback will be addressed :smile:

### CODE REVIEW

**Comment by arichiardi** [comment]
Working on the tests for this one

File: src/test/lumo/lumo/repl_tests.cljs [hunk 1]
```diff
@@ -103,7 +104,18 @@
...
+      (is-contains-completion "g/isF" "g/isFunction"))
+    (testing "Spec completions"
+      (testing "namespace fully qualified completion"
+        (s/def ::a-spec string?)
+        (is-contains-completion ":lumo.repl-tests/" ":lumo.repl-tests/a-spec")
+        (reset! s/registry-ref {}))
+      (testing "arbitrary fully qualified keyword"
+        (s/def :arbitrary/a-spec string?)
+        (is-contains-completion ":arbitrary/" ":arbitrary/a-spec")
+        (reset! s/registry-ref {})))))
```
Comment (anmonteiro) [inline src/test/lumo/lumo/repl_tests.cljs]: does it work for aliased nses?

e.g. `(s/def ::common/foo string?)` and contains a completion for `::common/f`

Comment (arichiardi) [reply src/test/lumo/lumo/repl_tests.cljs]: Gonna add a test for that

File: src/cljs/snapshot/lumo/repl.cljs [hunk 2]
```diff
@@ -1269,9 +1278,10 @@
 
 (defn- completion-candidates-for-current-ns []
   (let [cur-ns @current-ns]
-    (into (completion-candidates-for-ns cur-ns true)
-      (comp (mapcat keys) (map str))
-      ((juxt :renames :rename-macros :uses :use-macros) (get-namespace cur-ns)))))
+    (concat (completion-candidates-for-spec cur-ns)
+            (into (completion-candidates-for-ns cur-ns true)
```
Comment (arichiardi) [inline src/cljs/snapshot/lumo/repl.cljs]: I am not entirely sure we need this. It gets spec completions for the current namespace. What do you think?

Comment (anmonteiro) [reply src/cljs/snapshot/lumo/repl.cljs]: might be needed for `::a-spec`-style completions?

**Pull request merged**